lib/dx/evaluation.ex

```diff
@@ -37,7 +38,7 @@ defmodule Dx.Evaluation do
       {:debug, debug}, eval -> %{eval | debug?: debug}
       {:return_cache, return_cache}, eval -> %{eval | return_cache?: return_cache}
       {:args, args}, eval -> %{eval | args: Map.new(args)}
-      {key, val}, eval -> Map.replace!(eval, key, val)
+      {key, val}, eval -> %{eval | key => val}
```

Comment:
More idiomatic?

Reply:
Definitely more in line with the lines above 👍

As to idiomatic: I didn't even know this syntax works for dynamic keys. I can't remember having seen this in any repo so far. Just `grep`ped the projects I've cloned locally, and could find the syntax in many major repos though, so that's 👌 and I've learnt something again 🙂